edi-module/edi_translator.bal

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2023 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/io;
+
+type EDIContext record {|
+    EDISchema schema;
+    string[] ediText = [];
+    int rawIndex = 0;
+|};
+
+# Reads the given EDI text according to the provided schema
+#
+# + ediText - EDI text to be read
+# + schema - Schema of the EDI text
+# + return - JSON variable containing EDI data. Error if the reading fails.
+public isolated function read(string ediText, EDISchema schema) returns json|Error {
+    EDIContext context = {schema: schema};
+    EDIUnitSchema[] currentMapping = context.schema.segments;
+    context.ediText = splitSegments(ediText, context.schema.delimiters.segment);
+    EDISegmentGroup rootGroup = check readSegmentGroup(currentMapping, context, true);
+    return rootGroup.toJson();
+}
+
+# Writes the given JSON varibale into a EDI text according to the provided schema
```

Comment:
```suggestion
# Writes the given JSON varibale into a EDI text according to the provided schema.
```

Reply:
Best practice we are following is
```Always add a period (.) at the end of a function description. However, for the parameter and return type descriptions, omit the period if you have only one sentence. If there are multiple sentences, add the period at the end of each sentence. For example,```
Pls check the other places too.
https://ballerina.io/learn/generate-code-documentation/#write-ballerina-documentation